### DISCUSSION TITLE
How to define different classpaths for two main classes in the same module ?

### DISCUSSION THREAD

**Discussion by manuzhang** [sbt/sbt-native-packager#1401] [Q&A]
Current workaround

```scala
    bashScriptDefines := bashScriptDefines.value.map { v =>
      if (v.contains("app_classpath")) {
        """
         | if [ "$(basename "$0")" == "launcher" ]; then 
         |   declare -r app_classpath=...
         | else
         |   declare -r app_classpath=...
         | fi   
         |""".stripMargin
      } else {
        v
      }
    }
```

**Comment by manuzhang** [comment]
It seems people are not hanging out here. ping @muuki88 for some insight.

**Comment by muuki88** [comment]
Your workaround is actually okay. Another way to do it would be to configure two sbt sub modules and alter one of these settings

https://github.com/sbt/sbt-native-packager/blob/efe7c7a604ed8a93e283e7c58fe277790144ef53/src/main/scala/com/typesafe/sbt/packager/archetypes/JavaAppPackaging.scala#L59-L63

What's the use case for this? You are the first one trying this.
